pyproject.toml

```diff
@@ -0,0 +1,66 @@
+[build-system]
+requires = ["setuptools>=61.0"]  # Modern setuptools for static metadata
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "google-pso-data-validator"
+version = "5.0.0"  # Dynamic if using setuptools_scm, else keep static
+description = "A package to enable easy data validation"
+readme = "README.md"
+requires-python = ">=3.8"
+
+authors = [
+    {name = "PSO DVT Engineering team", email = "data-validator-eng@google.com"}
+]
+maintainers = [
+    {name = "PSO DVT Engineering team", email = "data-validator-eng@google.com"} 
+]
+
+license = {text = "Apache Software License"}
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "License :: OSI Approved :: Apache Software License",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Operating System :: OS Independent",
+]
+
+dependencies = [
+    "attrs==23.1.0",
+    "grpcio==1.53.2",
+    "lazy-object-proxy==1.9.0",
+    "marshmallow==3.19.0",
+    "fsspec>=2022.8.2",
+    "google-api-python-client==2.91.0",
+    "ibis-framework==5.1.0",
+    "impyla==0.18.0",
+    "SQLAlchemy==1.4.49",
+    "PyMySQL==1.1.1",
+    "psycopg2-binary==2.9.6",
+    "PyYAML==6.0",
+    "pandas==1.5.3",
+    "proto-plus==1.22.3",
+    "pyarrow==14.0.1",
+    "pydata-google-auth==1.8.0",
+    "google-cloud-bigquery==3.11.3",
+    "google-cloud-bigquery-storage==2.20.0",
+    "google-cloud-spanner==3.36.0",
+    "google-cloud-storage==2.10.0",
+    "jellyfish==1.0.0",
+    "tabulate==0.9.0",
+    "Flask==2.3.2",
+    "parsy==2.1",
+    "google-cloud-secret-manager>=2.16.0",
+]
+
+[project.optional-dependencies]
+apache-airflow = ["apache-airflow==1.10.11"]
+pyspark = ["pyspark==3.0.0"]
+
+[project.scripts]
+data-validation = "data_validation.__main__:main"
+
+[tool.setuptools]
+packages = ["data_validation", "data_validation.query_builder", "data_validation.result_handlers"]
```

Comment:
Should we ad the third_party/ibis packages here as well like we did in setup.py?
https://github.com/GoogleCloudPlatform/professional-services-data-validator/blob/develop/setup.py#L68